rke2-cloud-provider.yaml

```
@@ -0,0 +1,55 @@
+package:
+  name: rke2-cloud-provider
+  version: "1.33.1"
+  epoch: 0
+  description: cloud controller manager for rke2
+  copyright:
+    - license: Apache-2.0
+
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/rancher/image-build-rke2-cloud-provider
+      expected-commit: b24b25084b1c172f0b46a482c5dfca7987b9e2ed
+      # Needs to be updated manually from https://github.com/rancher/image-build-rke2-cloud-provider/releases
+      tag: v${{package.version}}-0.20250516163953-99d91538b132-build20250612
+
+  - uses: go/bump
+    with:
+      deps: |-
+        github.com/golang-jwt/jwt/v4@v4.5.2
+        golang.org/x/crypto@v0.35.0
+        golang.org/x/net@v0.38.0
+        golang.org/x/oauth2@v0.27.0
+
+  - name: Get k3s version
+    runs: |
+      K3S_VERSION=$(go list -f '{{.Version}}' -m github.com/k3s-io/k3s)
+
+  - uses: go/build
+    with:
+      packages: .
+      output: cloud-controller-manager
+      ldflags: -X github.com/k3s-io/k3s/pkg/version.Program=rke2 -X github.com/k3s-io/k3s/pkg/version.Version=$K3S_VERSION
```

```suggestion
      ldflags: -X github.com/k3s-io/k3s/pkg/version.Program=rke2 -X github.com/k3s-io/k3s/pkg/version.Version="${K3S_VERSION}"
```
If there are spaces or special characters in whatever comes back from that go list call, quoting like this is safer than unquoted variables.